src/Series.cpp

```diff
@@ -701,11 +701,16 @@ void Series::flushFileBased(
             switch (openIterationIfDirty(it->first, it->second))
             {
                 using IO = IterationOpened;
+            case IO::RemainsClosed:
+                // we might need to proceed further if the close status is
+                // ClosedInFrontend
+                // hence no continue here
+                // otherwise, we might forget to close files physically
+                break;
             case IO::HasBeenOpened:
+                // continue below
```

Comment:
todo: add iteration to `m_currentlyActiveIterations` here too
8cb746f8 Writing changes: Write current step(s) to snapshot attribute

Reply:
No idea what I meant here, this is the read mode, there's no need for this in read mode